src/lib.rs

```diff
@@ -291,6 +298,23 @@ impl SandboxFS {
         handles.get(&fh).expect("Kernel requested unknown handle").clone()
     }
 
+    /// Tracks a new file handle and assigns an identifier to it.
+    fn insert_handle(&mut self, handle: Arc<nodes::Handle>) -> u64 {
+        let fh = self.ids.next();
+        let mut handles = self.handles.lock().unwrap();
+        debug_assert!(!handles.contains_key(&fh));
+        handles.insert(fh, handle);
+        fh
+    }
+
+    /// Tracks a node, which may already be known.
+    fn insert_node(&mut self, node: Arc<nodes::Node>) {
+        let mut nodes = self.nodes.lock().unwrap();
+        if !nodes.contains_key(&node.inode()) {
```

Comment:
`nodes.entry(node.inode()).or_insert(node)`

Reply:
(This effectively can make a one-liner if you want it to be...)

Reply:
Done. Kept the function for consistency though.